test/OrchardCore.Tests/Modules/OrchardCore.OpenId/ApplicationControllerTests.cs

```diff
@@ -119,7 +119,13 @@ public async Task RedirectUrisAreValid(string uris, bool expectValidModel)
             model.AllowAuthorizationCodeFlow = true;
             model.RedirectUris = uris;
 
-            foreach (var validation in model.Validate(new ValidationContext(model)))
+            var validationContext = new ValidationContext(model);
+            var localizerMock = new Mock<IStringLocalizer<CreateOpenIdApplicationViewModel>>();
+            localizerMock.Setup(x => x[It.IsAny<string>(), It.IsAny<object[]>()])
+                .Returns((string name, object[] args) => new LocalizedString(string.Format(name, args), string.Format(name, args)));
```

Comment:
`Returns((name, args)`

Why `string.Format(name, args)` used twice for both resource name & value?

Reply:
Did not know exactly what resource and value meant in the context of localized string.

Reply:
The 'name` is the key that you looking for, while `value` is the returned localized string according to the specified `key`